test/e2e/setup/roll_out.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2023 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+maestro_latest_image=$(curl -s -X GET https://quay.io/api/v1/repository/redhat-user-workloads/maestro-rhtap-tenant/maestro/maestro | jq -s -c 'sort_by(.tags[].last_modified) | .[].tags[] | select(.name | test("sha256-|on-pr-|maestro-on-pull-request")| not) | .name' | head -n 1 | sed 's/\"//g')
+if [ $maestro_latest_image == "" ]; then
+    echo "Did not find the maestro latest image, exit!"
+    exit 1
+fi
+
+# if on service cluster
+if [ $(oc get deploy -n maestro -l app=maestro --no-headers | wc -l | sed 's/\ //g') -ge 1 ]; then
+    maestro_current_image=$(kubectl get deploy/maestro -n maestro -o jsonpath='{.spec.template.spec.containers[?(@.name=="service")].image}')
+    if [ $maestro_current_image == "" ]; then
+        echo "Did not find the maestro image, exit!"
+        exit 1
+    fi
+    image_repo=$(echo -n $maestro_current_image | awk -F ':' '{print $1}')
+    if [ $image_repo != $maestro_latest_image ]; then
+        # Roll out the maestro
+        kubectl set image deploy/maestro -n maestro *=$image_repo:$maestro_latest_image
+        kubectl rollout status deploy/maestro -n maestro --timeout=600s
+    fi
+fi
```

Comment:
Don’t we need to run the test cases before rolling out the agent?

Reply:
the test will not in the this roll-out scripts, I think it should be in the pipeline to call the test when the rollout finished.